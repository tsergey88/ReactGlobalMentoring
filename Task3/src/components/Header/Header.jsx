import React from "react";
import {Button} from "../Button";
import {RadioButton} from "../RadioButton";

export class Header extends React.Component {
  render() {
    return (
      <header>
        <div>NETFLIX ROULETTE</div>
        <div className="search-block">
          <h3>FIND YOUR MOVIE</h3>
          <input />
          <div>
            <div className="search-tags">
              <div><span>SEARCH BY</span></div>
              <div>
                <RadioButton title="TITLE" id="1" name="searchBy" defaultChecked />
                <RadioButton title="GENRE" id="2" name="searchBy" />
              </div>
            </div>
            <div><Button title="SEARCH" /></div>
          </div>
        </div>
      </header>
    )
  }
}