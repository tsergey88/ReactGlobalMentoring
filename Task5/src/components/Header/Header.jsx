import React from "react";

export class Header extends React.Component {
  render() {
    return (
      <header>
        <div className="logo">NETFLIX ROULETTE</div>
        {this.props.children}
      </header>
    );
  }
}
