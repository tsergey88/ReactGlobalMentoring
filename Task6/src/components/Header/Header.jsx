// @flow
import React from "react";

type Props = {
  children?: JSX.Element
};

export const Header = ({ children }: Props) => (
  <header>
    <div className="logo">NETFLIX ROULETTE</div>
    {children}
  </header>
);
