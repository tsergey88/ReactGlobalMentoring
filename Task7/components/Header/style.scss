.logo {
  display: flex;
  justify-content: space-between;
  margin-bottom: 1em;
  font-size: 24px;

  a {
    color: #e50914;
    text-decoration: none;
  }

  .btn {
    color: white;
    text-decoration: none;
  }
}