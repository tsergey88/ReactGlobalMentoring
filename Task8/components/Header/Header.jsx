// @flow
import React from "react";
import Link from "next/link";

import styles from "./style.scss";

type Props = {
  children?: JSX.Element,
  isFilmPage?: boolean
};

export const Header = ({ children, isFilmPage }: Props) => (
  <header>
    <div className={styles.logo}>
      <div>
        <Link href="/">
          <a>NETFLIX ROULETTE</a>
        </Link>
      </div>
      <div>
        {isFilmPage && (
          <Link href="/">
            <a className={styles.btn}>SEARCH</a>
          </Link>
        )}
      </div>
    </div>
    {children}
  </header>
);
