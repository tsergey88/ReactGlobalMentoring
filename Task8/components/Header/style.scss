:local(header) {
  color: #e50914;
  background: linear-gradient(45deg, rgba(103, 0, 31, .8), rgba(34, 101, 163, .5)), url(https://assets.nflxext.com/ffe/siteui/vlv3/eeca4431-e8de-4314-9a9f-189c87c2fe08/3702eb34-731a-4357-8750-20b0d9b98381/BY-en-20190325-popsignuptwoweeks-perspective_alpha_website_small.jpg);
  background-size: cover;
  padding: 10px 10em;

  :local(.logo) {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1em;
    font-size: 24px;

    a {
      color: #e50914;
      text-decoration: none;
    }

    .btn {
      color: white;
      text-decoration: none;
    }
  }
}