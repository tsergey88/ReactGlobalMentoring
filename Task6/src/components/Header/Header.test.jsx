import React from "react";
import { Header } from "./Header";
import { shallow } from "enzyme";

describe("Header component", () => {
  const component = shallow(<Header />);

  it("should be render correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should contain header", () => {
    expect(component.find("header")).toHaveLength(1);
  });

  it("should contain correct div", () => {
    expect(component.find("div")).toHaveLength(1);
  });
});
