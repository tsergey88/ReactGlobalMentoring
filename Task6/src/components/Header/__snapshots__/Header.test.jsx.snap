// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should be render correctly 1`] = `
<header>
  <div
    className="logo"
  >
    NETFLIX ROULETTE
  </div>
</header>
`;
